File: ios/Au10tixsdk.m
#import <React/RCTBridgeModule.h>
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import <React/RCTEventDispatcher.h>

@interface RCT_EXTERN_MODULE(Au10tixsdk, NSObject)

RCT_EXTERN_METHOD(prepareSdk:(id)data
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Au10tixsdk.swift
import Au10tixCore

@objc(Au10tixsdk)
class Au10tixsdk: NSObject {

   @objc(prepareSdk:withResolver:withRejecter:)
   func prepareSdk(token: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
        Au10tix.shared.prepare(with: token) { result in
        switch result {
        case .success(let sessionID):
            print("prepare success \(sessionID)")
            resolve(sessionID)
        case .failure(let error):
            print("prepare failure \(error.localizedDescription)")
        }
      }
   }
}
		
